components/script/dom/blob.rs

```diff
@@ -295,6 +299,43 @@ impl BlobMethods<crate::DomTypeHolder> for Blob {
         );
         p
     }
+
+    /// <https://w3c.github.io/FileAPI/#dom-blob-bytes>
+    fn Bytes(&self, can_gc: CanGc) -> Rc<Promise> {
+        let in_realm_proof = AlreadyInRealm::assert();
+        let p = Promise::new_in_current_realm(InRealm::Already(&in_realm_proof), can_gc);
+
+        // 1. Let stream be the result of calling get stream on this.
+        let stream = self.get_stream(can_gc);
+
+        // 2. Let reader be the result of getting a reader from stream.
+        //    If that threw an exception, return a new promise rejected with that exception.
+        match stream.and_then(|stream| stream.acquire_default_reader(can_gc)) {
+            Ok(reader) => {
+                // 3. Let promise be the result of reading all bytes from stream with reader.
+                reader.read_all_bytes(
+                    p.clone(),
+                    Rc::new(|promise, bytes| {
```

Comment:
Looking at step 3 and 4 of https://w3c.github.io/FileAPI/#dom-blob-bytes, I think it;s actually outdated(it should use `successSteps`)

So what you need to do is:
- Create the promise
- Pass a clone to both the success and failure steps(can you not just `move` them into the closure? If not, use a struct to implement those steps). 
- The success callback should take a bytes sequence, and then run the below steps.
- The failure steps should only reject the promise
- Call `read_all_bytes` with the two algorithms. 
- Return the promise below.